langchain4j-core/src/main/java/dev/langchain4j/agent/tool/ToolSpecifications.java

```diff
@@ -94,17 +154,15 @@ static Iterable<JsonSchemaProperty> toJsonSchemaProperties(Parameter parameter)
         if (type.isArray()) {
             return removeNulls(ARRAY, arrayTypeFrom(type.getComponentType()), description);
         }
-        if (Collection.class.isAssignableFrom(type)) {
-            return removeNulls(ARRAY, arrayTypeFrom(parameter.getParameterizedType()), description);
-        }
 
         if (type.isEnum()) {
             return removeNulls(STRING, enums((Class<?>) type), description);
         }
 
-        return removeNulls(OBJECT, description); // TODO provide internals
+        return null;
```

Comment:
All SimpleType are handled in `toJsonSchemaProperties(Class<?> type, JsonSchemaProperty description)`

If its Complex type we written null, so that it can be handled in ` toJsonSchemaProperties(Parameter parameter)` and `toJsonSchemaProperties(Field field, Set<Class<?>> visited)`